src/Microsoft.Restier.Core/Conventions/ConventionBasedEntitySetFilter.cs

```diff
@@ -22,23 +22,33 @@ private ConventionBasedEntitySetFilter(Type targetType)
             this.targetType = targetType;
         }
 
+        public IQueryExpressionFilter Inner { get; set; }
+
         /// <inheritdoc/>
         public static void ApplyTo(
             IServiceCollection services,
             Type targetType)
         {
             Ensure.NotNull(services, "services");
             Ensure.NotNull(targetType, "targetType");
-            services.CutoffPrevious<IQueryExpressionFilter>(new ConventionBasedEntitySetFilter(targetType));
+            services.ChainPrevious<IQueryExpressionFilter>(next => new ConventionBasedEntitySetFilter(targetType)
+            {
+                Inner = next,
+            });
         }
 
         /// <inheritdoc/>
         public Expression Filter(QueryExpressionContext context)
         {
             Ensure.NotNull(context, "context");
-            if (context.ModelReference == null)
+
+            if (Inner != null)
             {
-                return null;
+                var innerFilteredExpression = Inner.Filter(context);
+                if (innerFilteredExpression != null)
```

Comment:
Should be `innerFilteredExpression != context.VisitedNode` because later in `DefaultQueryHandler:
https://github.com/OData/RESTier/blob/master/src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs#L269-L281

If the returned node is different from the original one, the new one will be re-visited. But if the node is not changed, we can just go on with the original convention.